crates/matrix-sdk-base/src/store/traits.rs

```diff
@@ -808,6 +809,31 @@ pub enum StateStoreDataValue {
 
     /// A list of recently visited room identifiers for the current user
     RecentlyVisitedRooms(Vec<String>),
+
+    /// A composer draft for the room
+    ComposerDraft(ComposerDraft),
+}
+
+/// Struct that represents the current draft of the composer for the room
+#[derive(Debug, Clone, Serialize, Deserialize, uniffi::Record)]
+pub struct ComposerDraft {
+    /// The draft content in plain  text.
+    plain_text: String,
+    /// If the message is formatted in HTML, the HTML representation of the
+    /// message.
+    html_text: Option<String>,
+    /// The type of draft.
+    draft_type: DraftType,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, uniffi::Enum)]
+pub enum DraftType {
```

Comment:
```suggestion
pub enum ComposerDraftType {
```

so that we are sure it goes hand in hand with `ComposerDraft`?

Reply:
i don't see any other use case in which it is not tied to it, at least for now